src/use_case/add_project/AddProjectInteractor

```diff
@@ -0,0 +1,122 @@
+package use_case.add_project;
+import entity.Project;
+import entity.ProjectFactory;
+
+import java.io.*;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class AddProjectInteractor implements AddProjectInputBoundary {
+    final AddProjectDataAccessInterface userDataAccessObject;
+    final AddProjectOutputBoundary userPresenter;
+    final ProjectFactory projectFactory;
+
+    public AddProjectInteractor(AddProjectDataAccessInterface addProjectDataAccessInterface,
+                            AddProjectOutputBoundary addProjectOutputBoundary, ProjectFactory projectFactory) {
+        this.userDataAccessObject = addProjectDataAccessInterface;
+        this.userPresenter = addProjectOutputBoundary;
+        this.projectFactory =  projectFactory;
+    }
+
+    @Override
+    public void execute(AddProjectInputData addProjectInputData) {
+
+        String url = "https://api.openai.com/v1/chat/completions";
+        String apiKey;
+
+        try (BufferedReader br = new BufferedReader(new FileReader("../apikey.txt"))) {
+            apiKey = br.readLine();
+        } catch (IOException e) {
+            userPresenter.prepareFailView("Failed to fetch API key: " + e.toString());
+        }
+        String model = "gpt-3.5-turbo";
+
+        //A list of tasks. Each task is a list structured as follows: {Task Name, Task Desc, Task Deadline}
+        ArrayList<ArrayList<String>> tasks = new ArrayList<ArrayList<String>>();
+        StringBuilder tasksString = new StringBuilder();
+
+        try {
+            URL obj = new URI(url).toURL();
+            HttpURLConnection connection = (HttpURLConnection) obj.openConnection();
+            connection.setRequestMethod("POST");
+            connection.setRequestProperty("Authorization", "Bearer " + apiKey);
+            connection.setRequestProperty("Content-Type", "application/json");
+
+            // The request body
+            String body = "{\"model\": \"" + model + "\", \"messages\": " +
+                    "[{\"role\": \"user\", \"content\": \"Here are the details of a project I am working on: Project Title: " +
+                    addProjectInputData.getProjectTitle() + ". Project Description: " + addProjectInputData.getProjectDetails() +
+                    ". Project Deadline: " + addProjectInputData.getProjectDeadline() + ". Given this project, " +
+                    "break it down into smaller subtasks. I want your response formatted EXACTLY in the following " +
+                    "way: ;<insert task name>@ <insert brief task description>@ <insert task deadline as YYYY/MM/DD>; another task;" +
```

Comment:
SUGGESTION: change splitter from "@" to "|uwu|" for consistency and also to avoid potential issues (if the user uses "@" in their project name or description)? (also line 76)

Reply:
I will make these changes however I'm uncertain if this will confuse the API in the call prompt. Will continue to test and keep an eye on this new prompt.